core/src/avm1/activation.rs

```diff
@@ -240,18 +258,21 @@ impl<'gc> Activation<'gc> {
     }
 
     /// Resolve a particular named local variable within this activation.
+    ///
+    /// Because scopes are object chains, the same rules for `Object::get`
+    /// still apply here.
     pub fn resolve(
         &self,
         name: &str,
         avm: &mut Avm1<'gc>,
         context: &mut UpdateContext<'_, 'gc, '_>,
-    ) -> Value<'gc> {
+    ) -> Result<ReturnValue<'gc>, Error> {
         if name == "this" {
-            return Value::Object(self.this);
+            return Ok(Value::Object(self.this).into());
         }
 
         if name == "arguments" && self.arguments.is_some() {
-            return Value::Object(self.arguments.unwrap());
+            return Ok(Value::Object(self.arguments.unwrap()).into());
```

Comment:
Can these be set as normal references in the top of the scope chain instead of special cased here?